### PR TITLE
[CIVIS-8394] ENH logger for Civis Platform jobs

### PR DESCRIPTION
**Pull request by jacksonlee-civis**
Time for a new feature! This pull request adds a logger configured for Civis Platform jobs, with INFO-level (or below) logging showing up in black on the Civis Platform graphical user inteface, and WARNING-level (or above) logging in red. Internally, we've reinvented this same wheel so many times across codebases, so let's put this basic version in civis-python to help clean up downstream projects.

---

- [x] (For Civis employees only) Reference to a relevant ticket in the pull request title
- [x] Changelog entry added to `CHANGELOG.md` at the repo's root level
- [x] Description of change in the pull request description
- [x] If applicable, unit tests have been added and/or updated
- [x] The CircleCI builds have all passed

### CODE REVIEW

File: civis/loggers.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    Returns
+    -------
+    :class:`logging.Logger`
+    """
+    logger = logging.getLogger(
+        name if name is not None else globals()["__name__"]
+    )
+
+    if level is None:
+        logger.setLevel(os.getenv("CIVIS_LOG_LEVEL") or logging.INFO)
```
Comment (jacksonlee-civis) [inline civis/loggers.py:60]: I didn't know civis-python had been able to recognize this environment variable `CIVIS_LOG_LEVEL`, which was added for the CLI (https://github.com/civisanalytics/civis-python/pull/312). I'm using it more widely in this pull request and making it more visible in the documentation.

File: docs/source/user_guide.rst [hunk 2]
```diff
@@ -190,6 +190,40 @@ example, with pandas.
...
+.. code:: python
+
+    from civis import civis_logger
+
+    _LOG = civis_logger()  # instead of _LOG = logging.getLogger(__name__)
+
+    def some_function():
+        ...
+        _LOG.info("info level logging")  # appears in black on Civis Platform
+        _LOG.warning("something to look into potentially")  # appears in red
```
Comment (jacksonlee-civis) [inline docs/source/user_guide.rst:216]: I tested this on Civis Platform. The logging messages showed up as desired:

Screenshot 2024-05-09 at 3 27 20 PM

File: docs/source/user_guide.rst [hunk 3]
```diff
@@ -190,6 +190,40 @@ example, with pandas.
    >>> url = export_result.output[0].path
 
 
+Logging
+=======
```
Comment (jacksonlee-civis) [inline docs/source/user_guide.rst:194]: This is what this new section would look like in the compiled Sphinx docs:

Screenshot 2024-05-10 at 8 42 04 AM

**Review by Drew-Pappas — APPROVED** [review]
Wonderful PR! With the introduction of this it'll be much easier to to add in additional logging features like parameter obfuscation, downloadable logs, etc. without re-inventing the wheel over and over between projects.

LGTM! Super excited for this one

**Pull request merged**